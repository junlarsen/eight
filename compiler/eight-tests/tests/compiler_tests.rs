//! End-to-end tests for the compiler with a variety of different inputs and compiler options.

macro_rules! compiler_test_suite {
    ($name:ident, $directory:expr, [$($arg:expr),*]) => {
        #[test]
        fn $name() {
            use std::process::Command;

            insta::with_settings!({
                description => format!("compiler test suite for {}", stringify!($name)),
                snapshot_path => concat!(env!("CARGO_MANIFEST_DIR"), "/tests/snapshots/", stringify!($name)),
            }, {
                insta::glob!(concat!($directory, "/**/*.test"), |path| {
                    let mut cmd = Command::new(insta_cmd::get_cargo_bin("eightc"));
                    $(cmd.arg($arg);)*
                    let relative = path.strip_prefix(env!("CARGO_MANIFEST_DIR"))
                        .expect("failed to build relative path from Cargo.toml");
                    insta_cmd::assert_cmd_snapshot!(cmd.arg(relative));
                });
            })
        }
    }
}

compiler_test_suite!(ui, "ui", []);
compiler_test_suite!(syntax, "syntax", ["--emit-ast"]);
compiler_test_suite!(syntax_lowering, "syntax-lowering", ["--emit-hir"]);
compiler_test_suite!(type_inference, "type-inference", ["--emit-hir"]);
