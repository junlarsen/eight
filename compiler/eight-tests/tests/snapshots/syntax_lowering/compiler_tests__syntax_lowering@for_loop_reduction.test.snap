---
source: compiler/eight-tests/tests/compiler_tests.rs
description: compiler test suite for syntax_lowering
info:
  program: eightc
  args:
    - "--emit-hir"
    - tests/syntax-lowering/for_loop_reduction.test
input_file: compiler/eight-tests/tests/syntax-lowering/for_loop_reduction.test
snapshot_kind: text
---
success: false
exit_code: 1
----- stdout -----

----- stderr -----
Error: sema::trait_does_not_exist

  × trait Ord does not exist
   ╭─[tests/syntax-lowering/for_loop_reduction.test:2:19]
 1 │ fn loop_transform() {
 2 │   for (let i = 0; i < 10; i = i + 1) {
   ·                   ───┬──
   ·                      ╰── required trait Ord to exist
 3 │     let x = i;
   ╰────
